src/lib.rs

```diff
@@ -38,6 +38,14 @@
 use std::fmt::{self, Debug, Display, Formatter};
 use std::num::NonZeroUsize;
 
+/// Implement serde::Serialize and serde::Deserialize traits for Tree
```

Comment:
I do feel bad for nitpicking, again. But please bear with me.

I think this comment should be moved into `serde.rs` using the top-level `//!` syntax to concentrate all the Serde-related code and documentation in a single source file instead of lengthening `lib.rs`.

I also think it should be "proportional to tree depth" or "proportional to the depth of the tree".

Reply:
Comments moved, english fixed.